lib/spack/spack/util/environment.py

```diff
@@ -767,7 +767,7 @@ def from_sourcing_file(
 
         # Compute the environments before and after sourcing
         before = sanitize(
-            environment_after_sourcing_files(os.devnull, **kwargs),
+            environment_after_sourcing_files("cd %CD%", **kwargs),
```

Comment:
This should retain its cross platform compatibility (the current approach does not work on Linux/MacOS). Perhaps abstracting behind a function or module attribute would be ideal.

For context for other reviewers:
`os.devnul` isn't a value that's meaningful in this context on Windows, so we must affect some other no-op operation here. Changing directory to the current directory is an innocuous substitute.